### PR TITLE
AB#74552 Added AHP Financial landing page

### PR DESCRIPTION
**Pull request by rafalgajewski**

### CODE REVIEW

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investment.AHP.WWW/Views/Finance/Index.cshtml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+@model Guid
+@{
+    ViewData["Title"] = "Financial details";
+}
+<gds-div-grid-column-two-thirds>
+    <form asp-controller="Finance" asp-action="Start" asp-route-id="@Model"  asp-method="post" novalidate>
+        <gds-back-link href="@Url.Action("Index", "Home")" text="Back" class="govuk-!-margin-bottom-8">
+        </gds-back-link>
+
+        <gds-h1 class="govuk-heading-xl">
```
Comment (Carq) [inline src/HE.Investment.AHP.WWW/Views/Finance/Index.cshtml:11]: WCAGs?

**Review by Carq — APPROVED** [review]

File: src/HE.Investment.AHP.WWW/Views/Finance/Index.cshtml (hunk 1 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.WWW/Views/Finance/Index.cshtml:13]: I've added landmarks - WCAG check now passed fine

**Review by Carq — CHANGES_REQUESTED** [review]
Please remove commented code.

**Review by Carq — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/FinancialDetails/CommandHandlers/CreateFinancialDetailsCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+using HE.InvestmentLoans.Common.Validation;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualBasic;
+
+namespace HE.Investment.AHP.BusinessLogic.FinancialDetails.CommandHandlers;
+public class CreateFinancialDetailsCommandHandler : FinancialDetailsCommandHandlerBase, IRequestHandler<CreateFinancialDetailsCommand, OperationResult<CreateFinancialDetailsCommandResult>>
+{
+    private readonly IFinancialDetailsRepository _financialDetailsRepository;
+
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.Domain/FinancialDetails/CommandHandlers/CreateFinancialDetailsCommandHandler.cs:18]: @Carq are stylecop rules configured properly for this project?

File: src/HE.Investment.AHP.Domain/FinancialDetails/CommandHandlers/FinancialDetailsCommandHandlerBase.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        _repository = repository;
+        _logger = logger;
+    }
+
+    public async Task<OperationResult> Perform(Action<FinancialDetailsEntity> action, FinancialDetailsId financialDetailsId, CancellationToken cancellationToken)
+    {
+        //var userAccount = await _loanUserContext.GetSelectedAccount();
+
+        var financialScheme = await _repository.GetById(financialDetailsId, cancellationToken)
+            ?? throw new NotFoundException(nameof(FinancialDetails), financialDetailsId);
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.Domain/FinancialDetails/CommandHandlers/FinancialDetailsCommandHandlerBase.cs:29]: I can see that this exception is already thrown in repository (and this is correct)  so it'll never be null and throw is not needed here

File: src/HE.Investment.AHP.Domain/HE.Investment.AHP.Domain.csproj [hunk 4]
```diff
@@ -8,12 +8,21 @@
     </PropertyGroup>
 
     <ItemGroup>
-      <ProjectReference Include="..\HE.InvestmentLoans.BusinessLogic\HE.InvestmentLoans.BusinessLogic.csproj" />
-      <ProjectReference Include="..\HE.InvestmentLoans.Common\HE.InvestmentLoans.Common.csproj" />
+      <Folder Include="FinancialDetails\ValueObjects\" />
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.Domain/HE.Investment.AHP.Domain.csproj:11]: remove and fix formatting in line 18

File: src/HE.Investment.AHP.WWW/Views/Finance/Index.cshtml.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace HE.Investment.AHP.WWW.Views.Finance;
+
+public class Index : PageModel
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.WWW/Views/Finance/Index.cshtml.cs:5]: this file is not needed and can be removed

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.CRM.Plugins/Services/Accounts/AccountService.cs [hunk 6]
```diff
@@ -14,9 +14,9 @@ public class AccountService : CrmService, IAccountService
         private readonly IAccountRepository _accountRepository;
         private readonly IContactRepository _contactRepository;
 
-        private readonly string youRequested = "You requested";
-        private readonly string someoneElseRequested = "Someoneelse requested";
-        private readonly string noRequest = "No request";
+        private readonly string _youRequested = "You requested";
+        //private readonly string _someoneElseRequested = "Someoneelse requested";
```
[the remark this answers is not in the document]
Comment (rafalgajewski) [reply src/HE.CRM.Plugins/Services/Accounts/AccountService.cs]: commented code removed

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investment.AHP.Contract/HE.Investment.AHP.Contract.csproj [hunk 7]
```diff
@@ -20,10 +20,19 @@
...
 
   <ItemGroup>
+    <PackageReference Include="MediatR" Version="12.1.1" />
     <PackageReference Include="StyleCop.Analyzers" Version="1.2.0-beta.435">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
   </ItemGroup>
 
+  <ItemGroup>
```
[the remark this answers is not in the document]
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Contract/HE.Investment.AHP.Contract.csproj:30]: removed

File: src/HE.Investment.AHP.Domain/FinancialDetails/CommandHandlers/FinancialDetailsCommandHandlerBase.cs (hunk 3 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/FinancialDetails/CommandHandlers/FinancialDetailsCommandHandlerBase.cs]: removed

File: src/HE.Investment.AHP.Domain/HE.Investment.AHP.Domain.csproj (hunk 4 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/HE.Investment.AHP.Domain.csproj]: removed and fixed

File: src/HE.Investment.AHP.WWW/Views/Finance/Index.cshtml.cs [hunk 8]
```diff
@@ -0,0 +1,10 @@
+using Microsoft.AspNetCore.Mvc.RazorPages;
```
[the remark this answers is not in the document]
Comment (rafalgajewski) [reply src/HE.Investment.AHP.WWW/Views/Finance/Index.cshtml.cs]: no, it's already removed

File: src/HE.Investment.AHP.Domain/FinancialDetails/Repositories/FinancialDetailsRepository.cs [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+        var dataFromCookies = await Task.Run(() => _httpContextAccessor.HttpContext?.Request.Cookies[financialDetailsId.Value.ToString()]);
+        var result = JsonSerializer.Deserialize<FinancialDetailsEntity>(dataFromCookies ?? string.Empty) ?? throw new NotFoundException($"Financial details entity with id{financialDetailsId.Value} was not found.");
+        return result;
+    }
+
+    public async Task SaveAsync(FinancialDetailsEntity financialDetailsEntity, CancellationToken cancellationToken)
+    {
+        await Task.Run(() => _httpContextAccessor.HttpContext?.Response.Cookies.Append(financialDetailsEntity.Id.ToString(), JsonSerializer.Serialize(financialDetailsEntity)), cancellationToken);
+
+        //if (projectToSave.IsNewlyCreated)
```
[the remark this answers is not in the document]
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/FinancialDetails/Repositories/FinancialDetailsRepository.cs]: done

File: src/HE.Investment.AHP.Domain/FinancialDetails/CommandHandlers/CreateFinancialDetailsCommandHandler.cs [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+using HE.InvestmentLoans.Common.Validation;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualBasic;
+
+namespace HE.Investment.AHP.BusinessLogic.FinancialDetails.CommandHandlers;
+public class CreateFinancialDetailsCommandHandler : FinancialDetailsCommandHandlerBase, IRequestHandler<CreateFinancialDetailsCommand, OperationResult<CreateFinancialDetailsCommandResult>>
+{
+    private readonly IFinancialDetailsRepository _financialDetailsRepository;
+    //private readonly ILogger<CreateFinancialDetailsCommandHandler> _logger;
```
[the remark this answers is not in the document]
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/FinancialDetails/CommandHandlers/CreateFinancialDetailsCommandHandler.cs]: done

File: src/HE.Investment.AHP.WWW/Controllers/FinanceController.cs [hunk 11]
```diff
@@ -0,0 +1,39 @@
+using HE.Investment.AHP.Contract.Finance.Commands;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace HE.Investment.AHP.WWW.Controllers;
+
+[Route("finance")]
+public class FinanceController : Controller
```
[the remark this answers is not in the document]
Comment (rafalgajewski) [reply src/HE.Investment.AHP.WWW/Controllers/FinanceController.cs:8]: I think so, it will be added later on

**Pull request merged**